### ISSUE TITLE
MD038 Should catch code spans that start AND end with spaces

### ISSUE THREAD

**Issue by TheJaredWilcurt** [DavidAnson/markdownlint#473]
[MD038](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md038) - Spaces inside code span elements

Example code:

```md
caught: ` text`

caught: `text `

missed: ` text `

valid: `text`
```

Should work similar to [MD039](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md038) - Spaces inside link text

```md
[ caught](https://github.com)

[caught ](https://github.com)

[ caught ](https://github.com)

[valid](https://github.com)
```

**Comment by TheJaredWilcurt** [comment]
Also, this worked in version `0.22.0` and stopped working in `0.23.0`+.

* https://github.com/DavidAnson/markdownlint/compare/v0.22.0...v0.23.0

**Comment by DavidAnson** [comment]
Sorry about that! To be clear, I just released version 0.25.0 last night, but you are saying this broke two versions before that? Just coincidental timing, I guess?

**Comment by TheJaredWilcurt** [comment]
Oops, the version numbers are actually from `markdownlint-cli`

* https://github.com/igorshubovych/markdownlint-cli/compare/v0.22.0...v0.23.0

Which changed from `~0.19.0` to `~0.20.2` in it's deps.

**Comment by DavidAnson** [comment]
I couldn't figure out how I would've missed this for so long. It turns out, it's deliberate!

Here's the commit that allows it: https://github.com/DavidAnson/markdownlint/commit/bdc0246b34040c0e75b0acdd3a92ea4539f7d52b

The second bullet point here explains why: https://spec.commonmark.org/0.30/#code-spans
